src/net/wooga/jenkins/pipeline/BuildUtils.groovy

```diff
@@ -0,0 +1,56 @@
+package net.wooga.jenkins.pipeline
+
+import org.jenkinsci.plugins.pipeline.modeldefinition.Utils
+
+class BuildVersion {
+    final String version
+    final Boolean optional
+
+    BuildVersion(version, optional) {
+        this.version = version
+        this.optional = optional
+    }
+    
+    @Override
+    String toString() {
+        return "version : ${version}, optional : ${optional}"
+    }
+}
+
+// For each entry in a versions map, if they are not in the standard format
+// we process them into the newer format that supports optional arguments per version
+// (such as if it's optional, etc)
+// Example: '3001.x' -> [ version : '3001.x', optional : false ]
+def parseVersions(versions) {
+
+    final versionKey = "version"
+    final optionalKey = "optional"
+
+    versions = versions.collect { $it -> 
+    
+        def v = $it
```

Comment:
As I wrote in a PM I would first check if the value provided is of type closuere and execute it.

```groovy
if(v instanceof Closure) {
   v = v.call()
}
```

then continue with the check for Map.

Reply:
I admit at the time I didn't understand what you meant, but now I think I do? So this means that if a closure is provided within this collection it will return me a value, which then I can work with.

Reply:
correct  or allow lazy evalution.  It's not strictly  needed but I like to have it in.